src/main/java/uk/gov/dvsa/mot/framework/excel/ExcelDocument.java

```diff
@@ -0,0 +1,169 @@
+        addativeStyle.setFillForegroundColor(IndexedColors.LIGHT_GREEN.getIndex());
+        addativeStyle.setFillPattern(FillPatternType.SOLID_FOREGROUND);
+
+        Cell greenKey = sheet.createRow(2).createCell(0);
+        greenKey.setCellValue("Added to top ten");
+        greenKey.setCellStyle(addativeStyle);
+
+        CellStyle removedStyle = workbook.createCellStyle();
+        removedStyle.setFillForegroundColor(IndexedColors.ROSE.getIndex());
+        removedStyle.setFillPattern(FillPatternType.SOLID_FOREGROUND);
+
+        Cell redKey = sheet.createRow(3).createCell(0);
+        redKey.setCellValue("Removed from top ten");
+        redKey.setCellStyle(removedStyle);
+
+        int rowNumber = 4;
+        int indexCounter = 0;
+
+        for (ElasticSearchResult elasticSearchResult1 : resultSet1) {
+
+            ElasticSearchResult elasticSearchResult2 = resultSet2.get(indexCounter);
+            String searchTermData;
+
+            if (hasTheOrderChanged(elasticSearchResult1.getTopTenList(), elasticSearchResult2.getTopTenList())) {
+                searchTermData = String.format("%s, (%d, %d) Order has changed", elasticSearchResult1.getSearchTerm(),
+                        elasticSearchResult1.getTotalCount(), elasticSearchResult2.getTotalCount());
+            } else {
+                searchTermData = String.format("%s, (%d, %d)", elasticSearchResult1.getSearchTerm(),
+                        elasticSearchResult1.getTotalCount(), elasticSearchResult2.getTotalCount());
+            }
+
+            Row row = sheet.createRow(rowNumber++);
+
+            row.createCell(0).setCellValue(searchTermData);
+
+            int originalRow = rowNumber;
+
+            for (String result1 : elasticSearchResult1.getTopTenList()) {
+                Row topTenRow = sheet.createRow(originalRow++);
+
+                Cell originalValue = topTenRow.createCell(1);
+                originalValue.setCellValue(result1);
+
+                if (!elasticSearchResult2.getTopTenList().contains(result1)) {
+                    originalValue.setCellStyle(removedStyle);
+                }
+            }
+
+            int updatedRow = rowNumber;
+
+            for (String result2 : elasticSearchResult2.getTopTenList()) {
+                Row topTenRow;
+                Cell updatedValue;
+
+                try {
+                    topTenRow = sheet.getRow(updatedRow++);
+                    updatedValue  = topTenRow.createCell(2);
+                } catch (NullPointerException exception) {
+                    topTenRow = sheet.createRow(updatedRow++);
+                    updatedValue = topTenRow.createCell(2);
+                }
+
+                updatedValue.setCellValue(result2);
+
+                if (!elasticSearchResult1.getTopTenList().contains(result2)) {
+                    updatedValue.setCellStyle(addativeStyle);
+                }
+            }
+
+            rowNumber = Math.max(updatedRow, originalRow);
+            rowNumber++;
```

Comment:
Isn't this just the same as
rowNumber = Math.max(updatedRow, originalRow) + 1;

Reply:
squashed to one line